intellij/src/saros/intellij/ui/wizards/AddProjectToSessionWizard.java

```diff
@@ -177,15 +179,20 @@ public void next() {
          * @param project the project to create the module in
          * @param moduleName the name for the new module
          * @param moduleBasePath the base path for the new module
-         * @see AddProjectToSessionWizard#createModuleStub(String, Path, Project)
+         * @see AddProjectToSessionWizard#createBaseModule(String, String, Path, Project)
          */
         private void doNewModule(
             @NotNull Project project, @NotNull String moduleName, @NotNull Path moduleBasePath) {
 
+          Map<String, String> moduleParameters =
+              negotiation.getProjectNegotiationData(remoteProjectID).getAdditionalProjectData();
+
+          String moduleType = moduleParameters.get(ModuleConfigurationProvider.MODULE_TYPE_KEY);
```

Comment:
moduleType could theoretically be null.

Reply:
Jop. No idea how to handle this though. Will think about it.

Reply:
Done. Aborted negotiation if type not provided.